src/com/facebook/buck/swift/SwiftLibrary.java

```diff
@@ -210,9 +173,28 @@ SwiftCompile requireSwiftCompileRule(Flavor... flavors)
     return (SwiftCompile) rule;
   }
 
+  private BuildRule requireSwiftLinkRule(Flavor... flavors) throws NoSuchBuildTargetException {
+    BuildTarget requiredBuildTarget = getBuildTarget()
+        .withoutFlavors(ImmutableSet.of(SWIFT_LIBRARY_FLAVOR))
+        .withAppendedFlavors(CxxDescriptionEnhancer.SHARED_FLAVOR)
+        .withAppendedFlavors(flavors);
+    BuildRule rule = ruleResolver.requireRule(requiredBuildTarget);
+    if (!(rule instanceof CxxLink)) {
+      throw new RuntimeException(
+          String.format(
+              "Could not find CxxLink with target %s",
+              requiredBuildTarget));
+    }
+    return rule;
+  }
+
   @Override
   public NativeLinkable.Linkage getPreferredLinkage(CxxPlatform cxxPlatform) {
-    return Linkage.ANY;
+    if (getBuildTarget().getFlavors().contains(SWIFT_LIBRARY_FLAVOR)) {
+      return Linkage.STATIC;
```

Comment:
only companion swift library does contain `SWIFT_LIBRARY_FLAVOR`, we don't want to generate dylib for it but link statically swift source and objc in a mix target.

Maybe I should rename `SWIFT_LIBRARY_FLAVOR` to `SWIFT_COMPANION_FLAVOR`.